xrdsst/controllers/user.py

```diff
@@ -0,0 +1,195 @@
+import os
+import subprocess
+
+from cement import ex
+from xrdsst.controllers.base import BaseController
+from xrdsst.core.util import get_admin_credentials, get_ssh_key, get_ssh_user
+from xrdsst.resources.texts import texts
+
+
+class UserController(BaseController):
+    _GROUP_NAMES = ['xroad-security-officer',
+                    'xroad-registration-officer',
+                    'xroad-service-administrator',
+                    'xroad-system-administrator',
+                    'xroad-securityserver-observer']
+
+    class Meta:
+        label = 'user'
+        stacked_on = 'base'
+        stacked_type = 'nested'
+        description = texts['user.controller.description']
+
+    @ex(help="Create admin user", arguments=[])
+    def create(self):
+        self.create_user(self.load_config())
+
+    def create_user(self, conf):
+        for security_server in conf["security_server"]:
+            self.log_debug('Creating admin user for security server: ' + security_server['name'])
+            admin_credentials = get_admin_credentials(security_server, conf)
+            user_name = admin_credentials.split(":")[0]
+            pwd = admin_credentials.split(":")[1]
+            ssh_key = get_ssh_key(security_server, conf)
+            ssh_user = get_ssh_user(security_server, conf)
+            os_name = self.determine_os(ssh_key, ssh_user, security_server)
+            self.prepare_os(user_name, os_name, ssh_key, ssh_user, security_server)
+            self.add_groups(self._GROUP_NAMES, user_name, ssh_key, ssh_user, security_server)
+            self.add_user_with_groups(self._GROUP_NAMES, user_name, pwd, ssh_key, ssh_user, security_server)
+            self.log_info('Admin user \"' + user_name + '\" for security server ' + security_server['name'] +
+                          ' created.')
+
+    def determine_os(self, ssh_key, ssh_user, security_server):
+        self.log_debug('Determining operation system for security server: ' + security_server['name'])
+        os_name_cmd = 'ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "{}" {}@{} "{}"'.format(
+            ssh_key, ssh_user, self.security_server_address(security_server), "grep '^NAME' /etc/os-release")
+        if os.path.isfile(ssh_key):
+            try:
+                exitcode, data = subprocess.getstatusoutput(os_name_cmd)
+                if exitcode == 0:
+                    if data.find("Ubuntu") > 0:
```

Comment:
Arent these comparisons better off case insensitive?

Reply:
Made the comparisons case insensitive